lwrunner/runner.go

```diff
@@ -24,27 +24,85 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path"
 
-	"github.com/pkg/errors"
+	homedir "github.com/mitchellh/go-homedir"
 	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
 )
 
-// Exec executes a command on a remote host
-func Exec(host, cmd string) (string, error) {
-	user := "root"
+type Runner struct {
+	Hostname string
+	Port     int
+	*ssh.ClientConfig
+}
+
+func New(user, host string) *Runner {
+	// @afiune notify the user?
+	hostKeyCallback := ssh.InsecureIgnoreHostKey()
+
+	// try to use the known_hosts file as a host key callback
+	home, err := homedir.Dir()
+	if err == nil {
+		hostKeyFromKnownHosts, err := knownhosts.New(path.Join(home, ".ssh", "known_hosts"))
+		if err == nil {
+			hostKeyCallback = hostKeyFromKnownHosts
+		}
+	}
```

Comment:
Would this work in here instead?

```suggestion
	if home, err := homedir.Dir(); err == nil {
		hostKeyCallback, err = knownhosts.New(path.Join(home, ".ssh", "known_hosts"))
	}
```

Reply:
As talked over Slack, this misses the second error check. I will separate the logic into its own function to avoid the nested `if` statements.